### DISCUSSION TITLE
AWS Batch - optional outputs cause job to "exit 1" and fail

### DISCUSSION THREAD

**Discussion by PeterT2020** [broadinstitute/cromwell#7510] [Q&A]
When my batch jobs encounter an optional output that doesn't exist for a task, the delocalization block fails, setting DELOCALIZATION_FAILED=1 in the .scala. This is in the "aws-friendly" fork, which I believe is based off of Cromwell version 85. In the cromwell aws slack chat it has been mentioned that this issue has been encountered before and was potentially solved in later versions of cromwell.

Basically, what I want to ask is, does anyone know what fixed the issue? Can you point me towards the code that evaluates those optional outputs? I'm looking to patch the aws-friendly fork if possible, but looking through the diffs between 85=>87 was a brutal task, so I'm hoping someone can point me towards the direction I should be looking in.
